src/bowlingGame.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+class BowlingGame{
+
+    constructor() {
+        this.rolls = [];
+        this.currentRoll = 0;
+    }
+
+    roll(pins) {
+        this.rolls[this.currentRoll++] = pins;
+    }
+
+    score() {
+        let result = 0;
+        let rollIndex = 0;
+        let self = this;
```

Comment:
This is actually fine, and a trick that is used in JS regularly (often under the form `let that = this`).
Since some time this year, you can also have actual private methods in JS. See here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields

Reply:
I don't mind the way it's done here, however, In some ways it's even cleaner as you can be sure these methods are only ever used by score.
Another thing you could do instead of `let self = this;` is to have `let rolls = this.rolls` and then refer to `rolls` in the inner functions.

Reply:
Ah great! I couldn't think for the life of me why I couldn't just refer to this.rolls in my score method, as it kept saying undefined.  But I thought it was defined via the method it was in above.